invokeai/app/invocations/latent.py

```diff
@@ -0,0 +1,251 @@
+        width *= 8
+        height *= 8
+
+        dataURL = image_to_dataURL(image, image_format="JPEG")
+
+        context.services.events.emit_generator_progress(
+            context.graph_execution_state_id,
+            self.id,
+            {
+                "width": width,
+                "height": height,
+                "dataURL": dataURL
+            },
+            step,
+            self.steps,
+        )
+
+    def invoke(self, context: InvocationContext) -> LatentOutput:
+        def step_callback(state: PipelineIntermediateState):
+            self.dispatch_progress(context, state.latents, state.step)
+
+        model_info = context.services.model_manager.get_model(self.model)
+        model_name = model_info['model_name']
+        model_hash = model_info['hash']
+        model: StableDiffusionGeneratorPipeline = model_info['model']
+        model.scheduler = get_scheduler(
+            model=model,
+            scheduler_name=self.sampler_name
+        )
+        uc, c, extra_conditioning_info = get_uc_and_c_and_ec(self.prompt, model=model)
+
+        if isinstance(model, DiffusionPipeline):
+            for component in [model.unet, model.vae]:
+                configure_model_padding(component,
+                                        self.seamless,
+                                        self.seamless_axes
+                                        )
+        else:
+            configure_model_padding(model,
+                                    self.seamless,
+                                    self.seamless_axes
+                                    )
+
+        conditioning_data = ConditioningData(
+            uc,
+            c,
+            self.cfg_scale,
+            extra_conditioning_info,
+            postprocessing_settings=PostprocessingSettings(
+                threshold=0.0,#threshold,
+                warmup=0.2,#warmup,
+                h_symmetry_time_pct=None,#h_symmetry_time_pct,
+                v_symmetry_time_pct=None#v_symmetry_time_pct,
+            ),
+        ).add_scheduler_args_if_applicable(model.scheduler, eta=None)#ddim_eta)
+
+        noise = get_noise(self.width, self.height, model.device)
+
+        result_latents, result_attention_map_saver = model.latents_from_embeddings(
+            latents=torch.zeros_like(noise, dtype=torch_dtype()),
+            noise=noise,
+            num_inference_steps=self.steps,
+            conditioning_data=conditioning_data,
+            callback=step_callback
+        )
+
+        # https://discuss.huggingface.co/t/memory-usage-by-later-pipeline-stages/23699
+        torch.cuda.empty_cache()
+
+        name = f'{context.graph_execution_state_id}__{self.id}'
+        context.services.latents.set(name, result_latents)
+        return LatentOutput(
+            latent=LatentField(latent_name=name)
+        )
+
+
+# Latent to image
+class LatentToImageInvocation(BaseInvocation):
+    """Generates an image from a latent."""
```

Comment:
this is a nitpick, but before this gets baked into official terminology it might be worth renaming:

a "latent" isn't exactly the correct term. what these "latents" are is a tensor of floats that represents an encoding (=a very cleverly compressed version) of an image in the latent space of the VAE. 

in place of "latent" i'd propose "vae latent tensor" or just "latent tensor", or "vae latents", or at a pinch just "latents". my preference would be for `vae_latents`.

Reply:
I wasn't sure if this invocation in particular would work correctly (LatentToImage) - are the latents tied to the model that generated them and the vae it uses) at all?

Reply:
yes, for a given SD model the latent tensor generated by running diffusion using its unet will only look as "intended" when it's decoding using the VAE associated with that model.

this isn't a strict binding though - you'll still get coherent and even likely very acceptable output with a different VAE, but the results will be different, and this is usually not what's intended (although it does open creative possibilities).

Reply:
(note also that img2img also makes use of the VAE to turn pixels into a latent tensor - the same principle applies)

Reply:
I'm curious if there is a use case in which an image is encoded using one VAE and decoded using another? The fact that you can run the unet across a VAE that it wasn't trained on and get decent results suggests that this might be a useful thing to do.

Reply:
Oh I didn't know who to ask - is there a way to load just the vae but not the model? I presumably don't need the whole model when converting latents to images.

Reply:
@Kyle0654 according to https://huggingface.co/stabilityai/sd-vae-ft-mse you should be able to do something like:

```
from diffusers.models import AutoencoderKL
vae = AutoencoderKL.from_pretrained("stabilityai/sd-vae-ft-mse")
```

Reply:
if you have a `StableDiffusionPipeline` instantiated already you can just do `pipeline.vae`

Reply:
Yah, I'm grabbing it from the model, but I'm worried that it may pull a model back into memory (potentially from disk) when I really only need the vae. I'm not sure how large the vae is though, and if it needs caching in the model manager like the models.